### PR TITLE
Release 1.4.0

### PR DESCRIPTION
**Pull request by jeremiedbb**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jeremiedbb** [comment]
@conda-forge-admin, please rerender

**Review by jjerphan — APPROVED** [review]
LGTM modulo a :green_circle: CI.

Thank you for taking care of the release, @jeremiedbb and @glemaitre.

File: recipe/meta.yaml
Comment (jjerphan) [inline recipe/meta.yaml:38]: Actually, I would add this pin for safety with the release of numpy 2 which will happen soon.
```suggestion
    - numpy <2
```

**Review by jjerphan — CHANGES_REQUESTED** [review]
Actually, I think we need to be careful with the release of NumPy 2 and make sure that those builds only will work with NumPy < 2.

When NumPy 2 is release, we can then add build variant on NumPy on `main` and `rc` and add repodata patches for the builds of the previous releases of scikit-learn on conda-forge.

See #227.

File: recipe/meta.yaml
Comment (jeremiedbb) [reply recipe/meta.yaml:38]: Thanks for reminding us ! Unfortunately I already uploaded to PyPI :( So I'm going to fix the pyproject and regenerate the wheels and make a new tag, then I'll update this PR to point to the new tag and fix the numpy pinning.

Comment (glemaitre) [reply recipe/meta.yaml:38]: This is why I have the following comment above it:
https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

**Review by jjerphan — APPROVED** [review]
LGTM since the pin on numpy is already present.

**Comment by jeremiedbb** [comment]
Still the PR points to the wrong archive since we're making a new tag. I'll reopen when the tag is done.

**Pull request merged**
